app/src/main/java/com/android/periodpals/model/timer/TimerModelSupabase.kt

```diff
@@ -0,0 +1,65 @@
+package com.android.periodpals.model.timer
+
+import android.util.Log
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.postgrest.postgrest
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+private const val TAG = "TimerRepositorySupabase"
+private const val TIMERS = "timers"
+
+/**
+ * Implementation of TimerRepository using Supabase.
+ *
+ * @property supabaseClient The Supabase client used for making API calls.
+ */
+class TimerRepositorySupabase(private val supabaseClient: SupabaseClient) : TimerRepository {
+
+  override suspend fun loadTimer(onSuccess: (TimerDto) -> Unit, onFailure: (Exception) -> Unit) {
+    try {
+      withContext(Dispatchers.Main) {
+        val result =
+            supabaseClient.postgrest[TIMERS]
+                .select {}
+                .decodeSingle<TimerDto>() // RLS rules only allows user to check their own line
+        Log.d(TAG, "getTimer: Success")
+        onSuccess(result)
+      }
+    } catch (e: Exception) {
+      Log.d(TAG, "getTimer: fail to get timer: ${e.message}")
+      onFailure(e)
+    }
+  }
+
+  override suspend fun upsertTimer(
+      timerDto: TimerDto,
+      onSuccess: (TimerDto) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    try {
+      withContext(Dispatchers.Main) {
+        val result =
+            supabaseClient.postgrest[TIMERS].upsert(timerDto) { select() }.decodeSingle<TimerDto>()
+        Log.d(TAG, "upsertTimer: Success")
+        onSuccess(result)
+      }
+    } catch (e: Exception) {
+      Log.d(TAG, "upsertTimer: fail to upsert timer: ${e.message}")
+      onFailure(e)
+    }
+  }
+
+  override suspend fun deleteTimer(onSuccess: (TimerDto) -> Unit, onFailure: (Exception) -> Unit) {
+    try {
+      withContext(Dispatchers.Main) {
+        val result = supabaseClient.postgrest[TIMERS].delete { select() }.decodeSingle<TimerDto>()
+        Log.d(TAG, "deleteTimer: Success")
+        onSuccess(result)
+      }
+    } catch (e: Exception) {
+      Log.d(TAG, "deleteTimer: fail to delete timer: ${e.message}")
+      onFailure(e)
+    }
+  }
```

Comment:
This is perhaps correct use of Supabase's `delete` function but perhaps a more correct version would come if we added an id to `Timer` & `TimerDto` and added a filter just like the [Supbase docs](https://supabase.com/docs/reference/kotlin/delete) recommend.

Something like this:
```suggestion
  override suspend fun deleteTimer(
      timerID: String, 
      onSuccess: (TimerDto) -> Unit, 
      onFailure: (Exception) -> Unit
  ) {
    try {
      withContext(Dispatchers.Main) {
        supabaseClient.postgrest[TIMERS].delete { 
          filter {
            eq("id", timerId)
          }
        }
        Log.d(TAG, "deleteTimer: Success")
        onSuccess(result)
      }
    } catch (e: Exception) {
      Log.d(TAG, "deleteTimer: fail to delete timer: ${e.message}")
      onFailure(e)
    }
  }
```

Reply:
Okay. I had thought about that. However, since in the RLS policies, a user is only allowed to insert, select, update, delete the timer linked to its uid, I thought we wouldn't need it. Sure I can do the changes ! :)

![image](https://github.com/user-attachments/assets/f733268d-bc6f-4adf-a122-c07778f2b984)

Reply:
Did the changes !